internal/controller/ols_app_postgres_assets.go

```diff
@@ -0,0 +1,346 @@
+	postgresConfig := cr.Spec.OLSConfig.ConversationCache.Postgres
+	if postgresConfig.SharedBuffers == nil || postgresConfig.SharedBuffers.String() == "" {
+		cr.Spec.OLSConfig.ConversationCache.Postgres.SharedBuffers = &postgresSharedBuffers
+	}
+	if postgresConfig.MaxConnections == 0 {
+		cr.Spec.OLSConfig.ConversationCache.Postgres.MaxConnections = PostgresMaxConnections
+	}
+	defaultPermission := int32(0600)
+	tlsCertsVolume := corev1.Volume{
+		Name: "secret-" + PostgresCertsSecretName,
+		VolumeSource: corev1.VolumeSource{
+			Secret: &corev1.SecretVolumeSource{
+				SecretName:  PostgresCertsSecretName,
+				DefaultMode: &defaultPermission,
+			},
+		},
+	}
+	bootstrapVolume := corev1.Volume{
+		Name: "secret-" + PostgresBootstrapSecretName,
+		VolumeSource: corev1.VolumeSource{
+			Secret: &corev1.SecretVolumeSource{
+				SecretName: PostgresBootstrapSecretName,
+			},
+		},
+	}
+	configVolume := corev1.Volume{
+		Name: PostgresConfigMap,
+		VolumeSource: corev1.VolumeSource{
+			ConfigMap: &corev1.ConfigMapVolumeSource{
+				LocalObjectReference: corev1.LocalObjectReference{Name: PostgresConfigMap},
+			},
+		},
+	}
+	dataVolume := corev1.Volume{
+		Name: PostgresDataVolume,
+		VolumeSource: corev1.VolumeSource{
+			EmptyDir: &corev1.EmptyDirVolumeSource{},
+		},
+	}
+	volumes := []corev1.Volume{tlsCertsVolume, bootstrapVolume, configVolume, dataVolume, getPostgresCAConfigVolume()}
+	postgresTLSVolumeMount := corev1.VolumeMount{
+		Name:      "secret-" + PostgresCertsSecretName,
+		MountPath: OLSAppCertsMountRoot,
+		ReadOnly:  true,
+	}
+	bootstrapVolumeMount := corev1.VolumeMount{
+		Name:      "secret-" + PostgresBootstrapSecretName,
+		MountPath: PostgresBootstrapVolumeMount,
+		SubPath:   PostgresExtensionScript,
+		ReadOnly:  true,
+	}
+	configVolumeMount := corev1.VolumeMount{
+		Name:      PostgresConfigMap,
+		MountPath: PostgresConfigVolumeMount,
+		SubPath:   PostgresConfig,
+	}
+	dataVolumeMount := corev1.VolumeMount{
+		Name:      PostgresDataVolume,
+		MountPath: PostgresDataVolumeMount,
+	}
+	volumeMounts := []corev1.VolumeMount{postgresTLSVolumeMount, bootstrapVolumeMount, configVolumeMount, dataVolumeMount, getPostgresCAVolumeMount(path.Join(OLSAppCertsMountRoot, PostgresCAVolume))}
+	deployment := appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      PostgresDeploymentName,
+			Namespace: r.Options.Namespace,
+			Labels:    generatePostgresSelectorLabels(),
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: &cacheReplicas,
+			Selector: &metav1.LabelSelector{
+				MatchLabels: generatePostgresSelectorLabels(),
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: generatePostgresSelectorLabels(),
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:            PostgresDeploymentName,
+							Image:           r.Options.LightspeedServicePostgresImage,
+							ImagePullPolicy: corev1.PullIfNotPresent,
```

Comment:
we're using pull always for now until we're ready to produce sha-pinned image sets

Reply:
Changed as suggested.